assets/js/components/settings/SettingsActiveModule/Footer.js

```diff
@@ -168,6 +168,38 @@ export default function Footer( props ) {
 		);
 	}, [ slug, viewContext ] );
 
+	// Check if the resolution for the specified selector has finished.
+	// This allows us to determine if the data needed by the module is still being loaded.
+	// The primary reason for this loading check is to disable the submit button
+	// while the necessary data for the settings is still being loaded, preventing
+	// premature interactions by the user.
+	const isLoading = useSelect( ( select ) => {
+		const resolutionMapping = {
+			analytics: 'getAccounts',
+			tagmanager: 'getAccounts',
+			'search-console': 'getMatchedProperties',
+		};
+		const resolutionSelector = resolutionMapping[ slug ];
+
+		if ( ! module || ! resolutionSelector ) {
+			return false;
+		}
+
+		return ! select( module.storeName ).hasFinishedResolution(
+			resolutionSelector
+		);
+	} );
+
+	let buttonText;
+
+	if ( isSaving ) {
+		buttonText = __( 'Saving…', 'google-site-kit' );
+	} else if ( canSubmitChanges ) {
+		buttonText = __( 'Apply changes', 'google-site-kit' );
+	} else {
+		buttonText = __( 'Save', 'google-site-kit' );
+	}
```

Comment:
This would be a bit more straightforward, right?

```suggestion
	let buttonText = = __( 'Save', 'google-site-kit' );
	
	if ( canSubmitChanges ) {
		buttonText = __( 'Apply changes', 'google-site-kit' );
	}
	if ( isSaving ) {
		buttonText = __( 'Saving…', 'google-site-kit' );
	}
```

Reply:
Good catch 👍